src/main/java/de/qabel/desktop/DesktopClient.java

```diff
@@ -94,6 +108,52 @@ public static void main(String[] args) throws Exception {
         launch(args);
     }
 
+    private static ClientDatabase getConfigDatabase() {
+        boolean needsToMigrateLegacyDatabase = !Files.exists(DATABASE_FILE) && Files.exists(LEGACY_DATABASE_FILE);
+
+        try {
+            try {
+                connection = DriverManager.getConnection("jdbc:sqlite://" + DATABASE_FILE.toAbsolutePath());
+
+                try {
+                    try (Statement statement = connection.createStatement()) {
+                        statement.execute("PRAGMA FOREIGN_KEYS = ON");
+                    }
+                    ClientDatabase clientDatabase = new DesktopClientDatabase(connection);
+
+                    clientDatabase.migrate();
+
+                    // TODO cut below here after the config migration transition period (~ 03.05.2016)
+                    if (needsToMigrateLegacyDatabase) {
+                        logger.warn("found legacy database, migrating to new format");
+                        TransactionManager tm = new SqliteTransactionManager(connection);
+                        LegacyDatabaseMigrator legacyDatabaseMigrator = new LegacyDatabaseMigrator(
+                            new SQLitePersistence(LEGACY_DATABASE_FILE.toAbsolutePath().toString()),
+                            clientDatabase
+                        );
+                        tm.transactional(legacyDatabaseMigrator::migrate);
```

Comment:
Why is the deletion of the old database file not included here but instead outside of the outer try?